Various documentation and phpDoc fixes

crstauf opened the pull request:
Corrects several mistakes in the documentation, as well as in phpDoc blocks.

Fixes #453.
Fixes #469.

crstauf committed:
various doc and phpDoc fixes

crstauf committed:
include fix for #469

crstauf committed:
include fix for #453

rrennick commented on functions.php:
```diff
@@ -7,10 +7,10 @@
- * @param array  $args Arguments to pass when the hook triggers.
- * @param string $group The group to assign this job to.
- * @return string The action ID.
+ * @param string $hook The hook to trigger
```
The trailing periods should be left for this block to pass the sniffs.

crstauf replied:
There's no consistency on phpDoc blocks: some blocks [have periods](https://github.com/woocommerce/action-scheduler/blob/be0a27922b6ca34f5f5e924cb7265b78062a2918/functions.php#L10), [some do not](https://github.com/woocommerce/action-scheduler/blob/be0a27922b6ca34f5f5e924cb7265b78062a2918/functions.php#L22). 

If a decision is made on which should be used, do not mind making that happen.

rrennick replied:
> If a decision is made on which should be used, do not mind making that happen.

The current standard is ending period. But, the way PHPDocs formatting changes are addressed in any project I've worked on is to update a file when making other changes in that file.

crstauf replied:
@rrennick I've added periods in cdfbd42.

rrennick commented on docs/api.md:
```diff
@@ -158,12 +158,12 @@ as_unschedule_action( $hook, $args, $group )
 ### Return value
 
-(null)
+(string|null) The scheduled action ID if a scheduled action was found, or null if no matching action found.
```
I'm okay with changing the Action ID type to int so long as it's updated to that wherever we make changes.

crstauf replied:
I'm mostly interested in getting the documentation corrected in this PR. I would certainly prefer for this function to return `int` rather than `string`, and also prefer to leave that for another PR.

rrennick commented on functions.php:
```diff
@@ -24,7 +24,7 @@ function as_enqueue_async_action( $hook, $args = array(), $group = '' ) {
  * @param string $group The group to assign this job to
  *
- * @return string The job ID
+ * @return int The job ID
```
Let's update these to be `action ID` for consistency.

rrennick reviewed (changes requested):
@crstauf Thanks for submitting this. I've noted a few things to change before merging.

crstauf committed:
change "job ID" to "action ID"

https://github.com/woocommerce/action-scheduler/pull/474#pullrequestreview-367399480

crstauf commented:
@rrennick Responded to feedback, and pushed one commit.

crstauf committed:
add periods

rrennick reviewed (approved):
@crstauf Thanks for cleaning these up.

crstauf commented:
Welcome! 🙇
